### PR TITLE
RestTestImprovement

### PR DESCRIPTION
**Pull request by astyz**
1. Fixed about 150 warnings and improved code from RestTests
2. Added new REST Methods for SystemTime

Pull request status:
- [x] Tests added/updated
- [x] 80 % Coverage
- [x] Code conventions
- [x] Ready for review

### CODE REVIEW

File: plugins/weather/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/WeatherRestTest.java [hunk 1]
```diff
@@ -76,13 +79,13 @@ public void setUp() {
 	private void createDay() {
 		FIODataPoint p = Mockito.mock(FIODataPoint.class);
 		Mockito.when(p.summary()).thenReturn("Clear throughout the day");
-		Mockito.when(p.precipProbability()).thenReturn((double) 0);
+		Mockito.when(p.precipProbability()).thenReturn(Double.valueOf(0));
```
Comment (Legion2) [inline plugins/weather/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/WeatherRestTest.java]: use `0.0`

File: plugins/weather/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/WeatherRestTest.java [hunk 2]
```diff
@@ -76,13 +79,13 @@ public void setUp() {
...
 		Mockito.when(p.precipType()).thenReturn("no data");
-		Mockito.when(p.temperatureMin()).thenReturn((double) 15);
-		Mockito.when(p.temperatureMax()).thenReturn((double) 25);
+		Mockito.when(p.temperatureMin()).thenReturn(Double.valueOf(25));
```
Comment (Legion2) [inline plugins/weather/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/WeatherRestTest.java]: use `25.0`

File: plugins/weather/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/WeatherRestTest.java [hunk 3]
```diff
@@ -76,13 +79,13 @@ public void setUp() {
...
+		Mockito.when(p.temperatureMax()).thenReturn(Double.valueOf(25));
 		Mockito.when(p.sunriseTime()).thenReturn("05:00:00");
 		Mockito.when(p.sunsetTime()).thenReturn("21:00:00");
-		Mockito.when(p.timestamp()).thenReturn((long) 12345);
+		Mockito.when(p.timestamp()).thenReturn(Long.valueOf(12345));
```
Comment (Legion2) [inline plugins/weather/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/WeatherRestTest.java]: use `12345L`

File: plugins/weather/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/WeatherRestTest.java (hunk 1 above)
Comment (astyz) [reply plugins/weather/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/WeatherRestTest.java]: I know but then we would get warnings like "The expression of type double is boxed into Double". I think it was allowed to have this sort of warnings but if I do it with valueOf the warnings fall away. Should I change it anyway?

File: plugins/weather/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/WeatherRestTest.java (hunk 2 above)
Comment (astyz) [reply plugins/weather/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/WeatherRestTest.java]: see above

File: plugins/weather/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/WeatherRestTest.java (hunk 3 above)
Comment (astyz) [reply plugins/weather/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/WeatherRestTest.java]: see above

File: plugins/weather/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/WeatherRestTest.java (hunk 1 above)
Comment (Legion2) [reply plugins/weather/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/WeatherRestTest.java]: change the eclipse settings to ignore the warning

Comment (astyz) [reply plugins/weather/src/test/java/de/unistuttgart/iaas/amyassist/amy/plugin/weather/WeatherRestTest.java]: ok

**Pull request merged**